tqdm/_tqdm.py

```diff
@@ -390,22 +390,29 @@ def pandas(tclass, *targs, **tkwargs):
         deprecated_t = [tkwargs.pop('deprecated_t', None)]
 
         def inner_generator(df_function='apply'):
-            def inner(df, func, *args, **kwargs):
+            def inner(df, func, **kwargs):
                 """
                 Parameters
                 ----------
                 df  : (DataFrame|Series)[GroupBy]
                     Data (may be grouped).
                 func  : function
                     To be applied on the (grouped) data.
-                *args, *kwargs  : optional
+                **kwargs  : optional
                     Transmitted to `df.apply()`.
                 """
+                # *args intentionally not supported (see #244, #299)
+
                 # Precompute total iterations
                 total = getattr(df, 'ngroups', None)
                 if total is None:  # not grouped
-                    total = len(df) if isinstance(df, Series) \
-                        else df.size // len(df)
+                    if df_function == 'applymap':
+                        total = df.size
+                    else:
+                        axis = kwargs.get('axis', 0)
+                        total = df.shape[axis]
+                        if isinstance(df, DataFrame):
+                            total += 1  # pandas calls update once too many
```

Comment:
I think we need to get someone who develops similar things in pandas to take a look. Don't really have the time to investigate seeing as I don't even use this feature.